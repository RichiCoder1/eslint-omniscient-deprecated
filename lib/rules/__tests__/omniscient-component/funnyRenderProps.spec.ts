import { fix } from "./__setup__/setup";

describe("render conversion", () => {
    it("shouldn't handle rest argument as props", () => {
        const input = `
    import component from 'omniscient';

    const TestComponent = component((...test) => {
        return (<h1 {...test}>Test</h1>);
    })`;
        const report = fix(input);
        expect(report).not.toBeFixed();
    });
    it("should handle rest on props destructure", () => {
        const input = `
    import component from 'omniscient';

    const TestComponent = component(({ a , ...test}) => {
        return (<h1 {...test}>Test</h1>);
    })`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });
});
