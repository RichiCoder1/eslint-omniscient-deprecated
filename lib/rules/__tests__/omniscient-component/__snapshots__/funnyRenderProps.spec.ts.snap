// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render conversion should handle rest on props destructure 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

    import component from 'omniscient';

    class TestComponent extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        const { a, ...test } = this.props;
        return <h1 {...test}>Test</h1>;
    }
}",
}
`;
